### PR TITLE
CE-1195 Improve AutoFollow by moving it to an extension

### PR DESCRIPTION
**Pull request by adamkarminski**
To avoid running a time and resource consuming _maintenance/wikia/cronjobs/AutoFollow.php_ script it'd be better to automatically add a user to default watchlists after email confirmation. The new solution uses TaskRunner to perform the action of adding a user to watchlists and introduces new WF variable - wgAutoFollowWatchlist. You can use this variable of a PHP Array type to define watchlists on community wikias.

### CODE REVIEW

**Comment by timmyquivy** [comment]
Macbre & Adam, my understanding is that this pull request does not automatically enable this extension, but rather puts in place the infrastructure. In order to actually make this live, we would need to push an addition change to populate the watchlist array, yes? 

If so, I think we're okay with enabling. If not, if merging this code automatically starts subscribing people at the next code release, we need to NOT merge this. We have a legal issue regarding EU/JP opt-in laws we need to sort.

**Comment by adamkarminski** [comment]
Hey @timmyquivy!
This pull request won't turn the extension on. I'll wait until everything is sorted out to enable it.

File: extensions/wikia/AutoFollow/AutoFollow.setup.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	'es' => 3487,
+	'fi' => 3083,
+	'fr' => 10261,
+	'it' => 11250,
+	'ja' => 3439,
+	'nl' => 10466,
+	'pl' => 1686,
+	'pt' => 696403,
+	'ru' => 3321,
+	'uk' => 3321, // Add Ukrainian users to the blog in Russian
```
Comment (macbre) [inline extensions/wikia/AutoFollow/AutoFollow.setup.php]: Kind of up to date with the current international situation...

File: extensions/wikia/AutoFollow/AutoFollowTask.class.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		if ( $iFailures === 0 ) {
+			WikiaLogger::instance()->info(
+				"AutoFollow log: User {$sUserName} added to watchlist at {$wgSitename}.",
+				$aLogParams
+			);
+		} else {
+			WikiaLogger::instance()->info(
+				"AutoFollow log: {$iFailures} happened when adding {$sUserName} to watchlist at {$wgSitename}.",
+				$aLogParams
+			);
```
Comment (macbre) [inline extensions/wikia/AutoFollow/AutoFollowTask.class.php]: Let's log failures with higher logging level method (`error` perhaps?)

File: extensions/wikia/AutoFollow/AutoFollow.setup.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+/**
+ * @global Array The list of extension credits.
+ * @see http://www.mediawiki.org/wiki/Manual:$wgExtensionCredits
+ */
+$wgExtensionCredits['other'][] = array(
+	'path'              => __FILE__,
+	'name'              => 'AutoFollow',
+	'descriptionmsg'    => 'autofollow-ext-description',
```
Comment (macbre) [inline extensions/wikia/AutoFollow/AutoFollow.setup.php]: Is this message defined?

File: extensions/wikia/AutoFollow/AutoFollow.setup.php (hunk 1 above)
Comment (adamkarminski) [inline extensions/wikia/AutoFollow/AutoFollow.setup.php]: Yeah, I giggled myself when asked to make this change... :)

File: extensions/wikia/AutoFollow/AutoFollow.setup.php (hunk 3 above)
Comment (adamkarminski) [inline extensions/wikia/AutoFollow/AutoFollow.setup.php]: Nope, heh.

**Comment by adamkarminski** [comment]
ping @macbre

**Comment by macbre** [comment]
LGTM (TM)

**Pull request merged**